### PR TITLE
Closes #1794 Implemented compute_age_years().

### LINKED ISSUE #1794: Feature Request: Implement `compute_age_years()`
### Feature Idea

Implement `compute_age_years()`. It should correspond to `derive_var_age_years()`. `derive_var_age_years()` should be updated such that it calls the new function.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by DeclanHodges**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by DeclanHodges** [comment]
Function documentation:
image
image

**Comment by DeclanHodges** [comment]
`derive_var_age_years()` previously used two different methods for converting age to years, depending on whether `unit_var` was used to specify the age unit (method 1 below) or there was an AGEU variable in the dataset to use (method 2 below):
1) Convert ages to durations and then use `time_length()`.
2) Define the average duration of a year in each time unit and divide each age by the relevant duration.

From some testing, it appears as though both methods produce the same result. So, I have used method 1 (using `time_length()`) in the `compute_age_years()` function, which is now called in both of the above scenarios within `derive_vars_age_years()`. Please let me know if it is necessary to be using different methods in each scenario, but it looks fine to me to use `time_length()` for both.

File: R/compute_age_years.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#'   A numeric vector is expected.
+#'
+#' @param age_unit Age unit.
+#'
+#'   Either a string containing the time unit of all ages in `age` or a character
+#'   vector containing the time units of each age in `age` is expected. Note that
+#'   permitted values are cases insensitive (e.g. 'YEARS' is treated the same
+#'   as 'years' and 'Years').
+#'
+#'   Permitted Values: 'years', 'months', 'weeks', 'days', 'hours', 'minutes', 'seconds'.
```
Comment (bundfussr) [inline R/compute_age_years.R:16]: Please use backticks and double quote, e.g., `"YEARS"` instead of 'YEARS'.

File: R/compute_age_years.R [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#'
+#' @details Returns a numeric vector of ages in years. Note, underlying computations
+#' assume an equal number of days in each year (365.25).
```
Comment (bundfussr) [inline R/compute_age_years.R:19]: Could you add that the is double and not integer?
We added this to `derive_var_age_years()` due to user feedback.

File: R/compute_age_years.R [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#' @export
+#'
+#' @examples
+#' compute_age_years(
+#'   age = c(240, 360, 480),
+#'   age_unit = "MONTHS"
+#' )
+#'
+#' compute_age_years(
+#'   age = c(28, 1461, 10227),
```
Comment (bundfussr) [inline R/compute_age_years.R:36]: Could you add a value which results in a non integer result?

File: tests/testthat/test-compute_age_years.R [hunk 4]
```diff
@@ -0,0 +1,67 @@
+## Test 1: compute_age_years() works as expected when `age_unit` is a string ----
+test_that("compute_age_years Test 1: compute_age_years() works as expected when
+          `age_unit` is a string", {
```
Comment (bundfussr) [inline tests/testthat/test-compute_age_years.R:3]: Please avoid that the test description uses more than one line. Otherwise, the "Format test_that test file" addin does not work.

**Review by bundfussr — CHANGES_REQUESTED** [review]
Looks good!
I have just a few minor comments.

File: R/compute_age_years.R (hunk 2 above)
Comment (DeclanHodges) [reply R/compute_age_years.R:19]: Done.

File: R/compute_age_years.R (hunk 3 above)
Comment (DeclanHodges) [reply R/compute_age_years.R:36]: Done.

File: tests/testthat/test-compute_age_years.R (hunk 4 above)
Comment (DeclanHodges) [reply tests/testthat/test-compute_age_years.R:3]: Done.

**Comment by DeclanHodges** [comment]
Hi @bundfussr,
Thanks for your review! All of your comments are now addressed.

File: R/compute_age_years.R (hunk 1 above)
Comment (bundfussr) [reply R/compute_age_years.R]: Sorry, I was not clear. I meant to use `` `"YEARS"` ``. Then it is displayed as `"YEARS"`.

Comment (DeclanHodges) [reply R/compute_age_years.R]: Got it, thanks for clarifying. It now looks like this:
image

**Review by bundfussr — APPROVED** [review]

**Review by bundfussr — APPROVED** [review]
